incubator/telegraf/telegraf/templates/deployment.yaml

```diff
@@ -0,0 +1,39 @@
+# Describes a Kubernetes deployment of a container running Telegraf.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "telegraf.fullname" . }}
+  labels:
+    app: {{ template "telegraf.name" . }}
+    chart: {{ template "telegraf.chart" . }}
+    release: {{ .Release.Name }}
+    instance: {{ .Values.Instance | quote }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "telegraf.name" . }}
+      instance: {{ .Values.Instance | quote }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "telegraf.name" . }}
+        chart: {{ template "telegraf.chart" . }}
+        instance: {{ .Values.Instance | quote }}
+        release: {{ .Release.Name }}
+    spec:
+      hostNetwork: false
+      containers:
+        - name: telegraf
+          image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
```

Comment:
We don't allow image repository to be configurable, the image itself must be hard coded. If the image is trusted, we may allow configurable tags.